domain/crossmodelrelation/state/model/remoteapplication.go

```diff
@@ -72,6 +74,59 @@ func (st *State) AddRemoteApplicationOfferer(
 	return nil
 }
 
+// AddRemoteApplicationConsumer adds a new synthetic application representing
+// the remote relation on the consuming model, to this, the offering model.
+func (st *State) AddRemoteApplicationConsumer(
+	ctx context.Context,
+	applicationName string,
+	args crossmodelrelation.AddRemoteApplicationConsumerArgs,
+) error {
+	db, err := st.DB(ctx)
+	if err != nil {
+		return errors.Capture(err)
+	}
+
+	if err := db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		// Check if the application already exists.
+		if err := st.checkApplicationNameAvailable(ctx, tx, applicationName); err != nil {
+			return errors.Errorf("checking if application %q exists: %w", applicationName, err)
+		}
+
+		// Check the offer doesn't already exist.
+		if err := st.checkOfferDoesNotExist(ctx, tx, args.OfferUUID); err != nil {
+			return errors.Capture(err)
+		}
```

Comment:
Can you rename this method. This is actually `checkApplicationRemoteOffererDoesNotExist` - I know it's long, but it shows the intent. Essentially you need the equivalent of that, but for consumers - `checkApplicationRemoteConsumerDoesNotExist`.